libs/js-sdk/src/external-state-store.ts

```diff
@@ -0,0 +1,16 @@
+import { EffectiveFeatureValue } from '@featureboard/contracts'
+
+export interface ExternalStateStore {
+    /**
+     * Gets a stable copy of the feature values.
+     * Will only be used during the initialisation if retrieving the feature values from the API would fail.
+     * */
+    all(): Promise<Record<string, EffectiveFeatureValue['value'] | undefined>>
+    /**
+     * Update is called whenever an effective feature value is updated.
+     * It can be used to keep the external state store accurate.
+     * */
+    update(
+        store: Record<string, EffectiveFeatureValue['value'] | undefined>,
+    ): void | PromiseLike<any>
```

Comment:
Can prob remove the `void |` from here, i'd expect it to always be async now

Reply:
Do you see any issue with using same interface for JS SDK and Node SDK? 

ExternalStateStore<T>

Reply:
Nope